pkg/combustion/rpm.go

```diff
@@ -110,3 +104,48 @@ func writeRPMScript(ctx *image.Context, rpmFileNames []string) (string, error) {
 
 	return modifyRPMScriptName, nil
 }
+
+// determine whether RPM configuration is needed
+func skipRPMconfigre(ctx *image.Context) bool {
+	pkg := ctx.ImageDefinition.OperatingSystem.Packages
+
+	if isComponentConfigured(ctx, userRPMsDir) && len(pkg.AddRepos) > 0 {
```

Comment:
There are essentially two ways of specifying specific packages that need to be resolved for following this PR (the original code assumed that dependencies were already satisfied when placing RPM's into the `rpms/` directory):

1. Side-loaded RPM's placed into `rpms/`
2. Packages listed in the `packageList` (originally `pkgList`)

It's also possible for the user to specify `additionalRepos` for the resolver to look to find dependencies (or packages from the `packageList`). It's important that no matter whether you specify side-loaded packages or packages from a list, it always goes through the resolver so we can guarantee installation at boot time.

Reply:
You can have an RPM that has dependencies from a third party repository. Meaning that for instance if you want to add the `nvidia-utils-G06-545.29.02-150500.18.1.x86_64.rpm`, you would need to also add the `https://download.nvidia.com/suse/sle15sp5/` and `https://developer.download.nvidia.com/compute/cuda/repos/sles15/x86_64/` so that it can resolve its dependencies and install successfully. Otherwise if you only specify the RPM the install will fail with missing dependencies error.